src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -319,13 +405,23 @@ internal override bool GetMemberAndWriteJsonExtensionData(object obj, ref WriteS
             bool success;
             T value = Get!(obj);
 
+            if (ShouldSerialize != null)
```

Comment:
Can maybe collapse the two statements into one here?
```suggestion
            if (ShouldSerialize?.Invoke(obj, value) == false)
```

Reply:
Or just
```suggestion
            if (ShouldSerialize != null && !ShouldSerialize(obj, value))
```